src/probnum/filtsmooth/statespace/integrator.py

```diff
@@ -0,0 +1,265 @@
+        denominators = 2.0 * self.ordint + 1.0 - range[:, None] - range[None, :]
+        diffmat_1d = 1.0 / denominators
+        return np.kron(np.eye(self.spatialdim), self.diffconst ** 2 * diffmat_1d)
+
+    def transition_rv(self, rv, start, stop, **kwargs):
+        if not isinstance(rv, pnrv.Normal):
+            errormsg = (
+                "Closed form transitions in LTI SDE models is only "
+                "available for Gaussian initial conditions."
+            )
+            raise TypeError(errormsg)
+        step = stop - start
+        rv = self.precon.inverse(step) @ rv
+        rv, info = self.transition_rv_preconditioned(rv, start)
+        # print("does the cross-covariance have to be changed somehow??")
+        info["crosscov"] = self.precon(step) @ info["crosscov"] @ self.precon(step).T
+        return self.precon(step) @ rv, info
+
+    def transition_rv_preconditioned(self, rv, start, **kwargs):
+        return self.equivalent_discretisation_preconditioned.transition_rv(rv, start)
+
+    def transition_realization(self, real, start, stop, **kwargs):
+        if not isinstance(real, np.ndarray):
+            raise TypeError(f"Numpy array expected, {type(real)} received.")
+        step = stop - start
+        real = self.precon.inverse(step) @ real
+        real, info = self.transition_realization_preconditioned(real, start)
+        # print("does the cross-covariance have to be changed somehow??")
+        info["crosscov"] = self.precon(step) @ info["crosscov"] @ self.precon(step).T
+        return self.precon(step) @ real, info
+
+    def transition_realization_preconditioned(self, real, start, **kwargs):
+        return self.equivalent_discretisation_preconditioned.transition_realization(
+            real, start
+        )
+
+    def discretise(self, step):
+        """
+        Equivalent discretisation of the process.
+
+        Overwrites matrix-fraction decomposition in the super-class.
+        Only present for user's convenience and to maintain a clean interface.
+        Not used for transition_rv, etc..
+        """
+
+        dynamicsmat = (
+            self.precon(step)
+            @ self.equivalent_discretisation_preconditioned.dynamicsmat
+            @ self.precon.inverse(step)
+        )
+        diffmat = (
+            self.precon(step)
+            @ self.equivalent_discretisation_preconditioned.diffmat
+            @ self.precon(step).T
+        )
+        zero_force = np.zeros(len(dynamicsmat))
+        return discrete_transition.DiscreteLTIGaussian(
+            dynamicsmat=dynamicsmat, forcevec=zero_force, diffmat=diffmat
+        )
+
+
+class IOUP(Integrator, sde.LTISDE):
+    """Integrated Ornstein-Uhlenbeck process in :math:`d` dimensions."""
+
+    def __init__(
+        self,
+        ordint: int,
+        spatialdim: int,
+        driftspeed: float,
+        diffconst: float,
+    ):
+        # Other than previously in ProbNum, we do not use preconditioning for IOUP by default.
```

Comment:
are you okay with this? I can use the `TaylorCoordinates` as a preconditioner here, too, but thought it might be cleaner to leave the preconditioner for IOUP, etc., open ended.

Reply:
I don't have a real opinion on this, I never left the comfort zone of the IBM prior. Whatever you think is best here :)

Reply:
Lets not have preconditioning until it bites us then